src/components/Table/EditColumnsButton.tsx

```diff
@@ -0,0 +1,90 @@
+import { useMenuTriggerState } from "react-stately";
+import { CheckboxGroup, CheckboxGroupItemOption } from "src/inputs";
+import { Css } from "../../Css";
+import { useTestIds } from "../../utils";
+import { Button } from "../Button";
+import { isIconButton, isTextButton, OverlayTrigger, OverlayTriggerProps } from "../internal/OverlayTrigger";
+import { GridColumn, Kinded } from "./GridTable";
+import { GridColumnState } from "./useColumns";
+
+interface EditColumnsButtonProps<R extends Kinded, S>
+  extends Pick<OverlayTriggerProps, "trigger" | "placement" | "disabled" | "tooltip"> {
+  columns: GridColumn<R, S>[];
+  setColumns: Dispatch<SetStateAction<GridColumnState<R, S>>>;
+  title?: string;
+  // for storybook purposes
+  defaultOpen?: boolean;
+}
+
+export function EditColumnsButton<R extends Kinded, S = {}>(props: EditColumnsButtonProps<R, S>) {
+  const { defaultOpen, disabled, columns, setColumns, trigger, title } = props;
+  const state = useMenuTriggerState({ isOpen: defaultOpen });
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  const { menuTriggerProps } = useMenuTrigger({ isDisabled: !!disabled }, state, buttonRef);
+  const tid = useTestIds(
+    props,
+    isTextButton(trigger) ? trigger.label : isIconButton(trigger) ? trigger.icon : trigger.name,
+  );
+
+  const editableColumns = useMemo(
+    () =>
+      columns.filter((column) => {
+        if (!column.canHide) return false;
+        if (!column.name || column.name.length === 0) {
+          console.warn("Column is missing 'name' property required by the Edit Columns button", column);
+          return false;
+        }
+        return true;
+      }),
+    [columns],
+  );
+
+  const selectedColumns: string[] = editableColumns
+    .map((column) => {
+      if (column.canHide && column.visible) return column.name;
+    })
+    .filter((column) => !!column) as string[];
+
+  const options: CheckboxGroupItemOption[] = editableColumns.map((column) => ({
+    label: capitalCase(column.name!)!,
+    value: column.name!,
+  }));
+
+  const [selectedValues, setSelectedValues] = useState<string[]>(selectedColumns ?? []);
+
+  const clearSelections = useCallback(() => {
+    // When clearing all selections, all hide-able columns should be filtered out of `visibleColumns`.
+    setColumns((prevState) => ({ ...prevState, visibleColumns: columns.filter((column) => !column.canHide) }));
+    setSelectedValues([]);
+  }, [columns]);
+
+  useEffect(() => {
+    setColumns((prevState) => ({
+      ...prevState,
+      visibleColumns: columns.filter((column) => (column.canHide ? selectedValues.includes(column.name!) : true)),
+    }));
+  }, [selectedValues]);
+
+  return (
+    <OverlayTrigger {...props} menuTriggerProps={menuTriggerProps} state={state} buttonRef={buttonRef} {...tid}>
+      <div
+        css={{
+          ...Css.bgWhite.py5.px3.maxwPx(380).dg.gap1.$,
+          "&:hover": Css.bshHover.$,
+        }}
+      >
+        <CheckboxGroup
```

Comment:
@bdow , is there a way to set `xsSb` to the `CheckboxGroup` label?

Reply:
Not at the moment. I would suggest maybe adding in a `hideLabel?: boolean` property. 
```
<CheckboxGroup label="Columns" hideLabel />
```
Then you can put your own "label" with the desired styles. But you'd still define the label for accessibility purposes, it just wouldn't show.